src/main/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -11,152 +12,188 @@
-        }
-    }
-
     /**
-     * create the internal JexlContext, only when required.  This code is where new JEXL context variables
-     * should get added.
-     *
+     * Note: due to laziness, this accessor actually modifies the instance by possibly forcing evaluation of an Jexl expression.
+     * @throws IllegalArgumentException when {@code o} is {@code null}
      */
-    private void createContext() {
-        if ( vc == null ) {
-            jContext = new MapContext(Collections.emptyMap());
-        }
-        else if (g == null) {
-            jContext = new VariantJEXLContext(vc);
-        }
-        else {
-            jContext = new GenotypeJEXLContext(vc, g);
+    public Boolean get(Object o) {
+        if(o == null){
+            throw new IllegalArgumentException("Query key is null");
         }
-    }
-
-    /**
-     * @return the size of the internal data structure
-     */
-    public int size() {
-        return jexl.size();
-    }
-
-    /**
-     * @return true if we're empty
-     */
-    public boolean isEmpty() { return this.jexl.isEmpty(); }
-
-    /**
-     * do we contain the specified key
-     * @param o the key
-     * @return true if we have a value for that key
-     */
-    public boolean containsKey(Object o) { return jexl.containsKey(o); }
 
-    public Boolean get(Object o) {
         // if we've already determined the value, return it
         if (jexl.containsKey(o) && jexl.get(o) != null) return jexl.get(o);
 
-        // try and cast the expression
-        JexlVCMatchExp e = (JexlVCMatchExp) o;
+        // otherwise cast the expression and try again
+        final JexlVCMatchExp e = (JexlVCMatchExp) o;
         evaluateExpression(e);
         return jexl.get(e);
     }
 
     /**
-     * get the keyset of map
-     * @return a set of keys of type JexlVCMatchExp
+     * do we contain the specified key
+     * @param o the key
+     * @return true if we have a value for that key
      */
+    public boolean containsKey(Object o) { return jexl.containsKey(o); }
+
     public Set<JexlVCMatchExp> keySet() {
         return jexl.keySet();
     }
 
     /**
-     * get all the values of the map.  This is an expensive call, since it evaluates all keys that haven't
-     * been evaluated yet.  This is fine if you truely want all the keys, but if you only want a portion, or  know
+     * Get all the values of the map, i.e. the {@link Boolean} values.
+     * This is an expensive call, since it evaluates all keys that haven't been evaluated yet.
+     * This is fine if you truly want all the keys, but if you only want a portion, or  know
      * the keys you want, you would be better off using get() to get them by name.
+     *
+     * Note: due to laziness, this accessor actually modifies the instance by possibly forcing evaluation of an Jexl expression.
      * @return a collection of boolean values, representing the results of all the variants evaluated
+     *
+     * @throws IllegalArgumentException
```

Comment:
Can you describe when it throws IllegalArgumentException, which is strange since it takes no arguments.